### ISSUE TITLE
`assign()` for new parameters

### ISSUE THREAD

**Issue by markvdw** [GPflow/GPflow#1191]
I just saw @st--'s stackoverflow post, where it was written "In the MWE you assign a new value to a Parameter that is already existing (and does not have the logistic transform)". I was doing exactly this the other day, since in [models.ipynb](https://github.com/GPflow/GPflow/blob/b0e09ee6d669b2761cb14778507a903c38a9e0ff/doc/source/notebooks/understanding/models.ipynb) cell 10 (below), it recommends to assign a `Parameter` object using `.assign()`.

Is this an error in the documentation? It did not work for me, so I changed it to ` = `, but I didn't follow up on it any further.

```
old_parameter = m.kernel.kernels[0].lengthscale
new_parameter = gpflow.Parameter(old_parameter.value(),
                                 trainable=old_parameter.trainable,
                                 prior=old_parameter.prior,
                                 name=old_parameter.name.split(':')[0],  # tensorflow is weird and adds ':0' to the name
                                 transform=tfp.bijectors.Exp())
m.kernel.kernels[0].lengthscale.assign(new_parameter)
```

**Comment by awav** [comment]
Hello, @markvdw, this is not a bug. The `assign` were never supposed to work with Parameter, it is a part of the Tensor interface. You are doing everything right with replacing a parameter with an equal sign. Just to clarify: `=` sign is a setter for the module's attributes and `assign` is a part of `tf.Variable` interface to be able to change value of the variable/parameter.

**Comment by st--** [comment]
It's a bug in the documentation, which is misleading.

**Comment by awav** [comment]
The issue addressed at #1193
